src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp

```diff
@@ -110,168 +154,302 @@ class boss_thaddius : public CreatureScript
+                                if (stalagg->AI())
+                                    stalagg->AI()->DoAction(ACTION_TRANSITION_3);
+
+                            _phase = PHASE_THADDIUS;
+                            events.SetPhase(PHASE_THADDIUS);
+
+                            Talk(SAY_AGGRO);
+                            DoCastAOE(SPELL_POLARITY_SHIFT);
+
+                            events.ScheduleEvent(EVENT_ENABLE_BALL_LIGHTNING, 10 * IN_MILLISECONDS, 0, PHASE_THADDIUS);
+                            events.ScheduleEvent(EVENT_SHIFT, 30 * IN_MILLISECONDS, 0, PHASE_THADDIUS);
+                            events.ScheduleEvent(EVENT_CHAIN, urand(10, 20) * IN_MILLISECONDS, 0, PHASE_THADDIUS);
+                            events.ScheduleEvent(EVENT_BERSERK, 6 * MINUTE * IN_MILLISECONDS, 0, PHASE_THADDIUS);
+
+                            me->GetMotionMaster()->Clear();
+                            me->GetMotionMaster()->MoveIdle();
+                            break;
+                        case EVENT_ENABLE_BALL_LIGHTNING:
+                            ballLightningEnabled = true;
+                            break;
+                        case EVENT_SHIFT:
+                            me->CastStop(); // shift overrides all other spells
+                            DoCastAOE(SPELL_POLARITY_SHIFT);
+                            events.ScheduleEvent(EVENT_SHIFT, 30 * IN_MILLISECONDS, PHASE_THADDIUS);
+                            break;
+                        case EVENT_CHAIN:
+                            if (me->FindCurrentSpellBySpellId(SPELL_POLARITY_SHIFT)) // delay until shift is over
+                                events.ScheduleEvent(EVENT_CHAIN, 3 * IN_MILLISECONDS, 0, PHASE_THADDIUS);
+                            else
+                            {
+                                DoCastVictim(SPELL_CHAIN_LIGHTNING);
+                                events.ScheduleEvent(EVENT_CHAIN, urand(10, 20) * IN_MILLISECONDS, PHASE_THADDIUS);
+                            }
+                            break;
+                        case EVENT_BERSERK:
+                            DoCast(me, SPELL_BERSERK);
+                            break;
+                    }
                 }
+
+                if (_phase == PHASE_THADDIUS)
+                    if (me->IsWithinMeleeRange(me->GetVictim()))
+                        DoMeleeAttackIfReady();
+                    else
+                        if (ballLightningEnabled)
+                            DoCastVictim(SPELL_BALL_LIGHTNING);
             }
 
-            if (events.GetTimer() > 15000 && !me->IsWithinMeleeRange(me->GetVictim()))
-                DoCastVictim(SPELL_BALL_LIGHTNING);
-            else
-                DoMeleeAttackIfReady();
-        }
+        private:
+            Phases _phase;
+            bool stalaggAlive;
+            bool feugenAlive;
+            bool ballLightningEnabled;
+            bool shockingEligibility;
```

Comment:
None of these have been initialized (in the constructor or anywhere), not even on Reset()